packages/thyra/core/src/interfaces/nonlinear/model_evaluator/fundamental/Thyra_ModelEvaluator.hpp

```diff
@@ -728,6 +804,9 @@ class ModelEvaluator : public ModelEvaluatorBase {
   /** \brief Return the vector space for the state function <tt>f(...) <: RE^n_x</tt>. */
   virtual RCP<const VectorSpaceBase<Scalar> > get_f_space() const = 0;
 
+  /** \brief Return the dual vector space for the state function <tt>f(...) <: RE^n_x</tt>. */
```

Comment:
Should this not be per virtual and have the default implementation moved to the class `Thyra::ModelEvaluatorDefaultBase`?  This base class is supposed to be a pure virtual base class.

Reply:
Thanks!